File: modules/react-native-nitro-lab/ios/HybridLab.swift
import Foundation
import NitroModules

// `npx nitrogen` 후 생성되는 HybridLabSpec 을 상속.
class HybridLab: HybridLabSpec {
  func hello(name: String) throws -> String {
    return "Hello, \(name)! (from Nitro/Swift)"
  }

  func add(a: Double, b: Double) throws -> Double {
    return a + b
  }

  func delay(ms: Double) throws -> Promise<Void> {
    return Promise.async {
      try await Task.sleep(nanoseconds: UInt64(ms) * 1_000_000)
    }
  }
}

File: modules/react-native-nitro-lab/android/src/main/java/com/margelo/nitro/nitrolab/HybridLab.kt
package com.margelo.nitro.nitrolab

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.core.Promise

// `npx nitrogen` 후 생성되는 HybridLabSpec 을 상속.
@Keep
@DoNotStrip
class HybridLab : HybridLabSpec() {
  override fun hello(name: String): String = "Hello, $name! (from Nitro/Kotlin)"

  override fun add(a: Double, b: Double): Double = a + b

  override fun delay(ms: Double): Promise<Unit> = Promise.async {
    kotlinx.coroutines.delay(ms.toLong())
  }
}
